ast/builtins.go

```diff
@@ -1039,6 +1040,17 @@ var Sprintf = &Builtin{
  * Numbers
  */
 
+// RandIntn returns a random number 0 - n
+var RandIntn = &Builtin{
+	Name: "rand.intn",
+	Decl: types.NewFunction(
+		types.Args(
```

Comment:
Let's change the function signature from `(number) => number` to `(string, number) => number`.

Reply:
Why?

Reply:
To be more specific, is `type.S` some kind of implied var?  Looking in that file I see types with (number, number) => number.

Reply:
Built-in functions must be idempotent. We could omit the string parameter but that would preclude policies from generating multiple random values within the same range. To allow policies to generate multiple values within the same range we can include an extra string parameter that serves as an identifier for the random value. See my comment below that links to the uuid.rfc4122 function that does the same thing.

Reply:
Is the seed on the built in context consistent between built in calls?  If that's the case seeding rand with it would make it consistent on number of calls.

Reply:
Using the seed like that implicitly relies on ordering. There's no ordering guarantee in Rego and there's no guarantee on the number of times the expression would be evaluated by OPA. The only way to guarantee determinism in this case is to memoize the result.

Reply:
@tsandall @srenatus Still no string arg updated with the cache and new calling format.  Will add that next.